File: appLib/src/main/java/com/jsx/applib/utils/Param.java
package com.jsx.applib.utils;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Author: JackPan
 * Date: 2021-10-20
 * Time: 10:45
 * Description:
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Param {
}

File: appLib/src/main/java/com/jsx/applib/utils/Param.kt
package com.jsx.applib.utils

/**
 * Author: JackPan
 * Date: 2021-10-23
 * Time: 14:03
 * Description: 参数解析注解
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class Param(val value:String)